ivi-layermanagement-examples/EGLWLMockNavigation/src/Car.cpp

```diff
@@ -93,9 +93,16 @@ Car::Car(vec3f position, vec3f size, vec4f color, ShaderTexture* shader, Texture
 
 void Car::render()
 {
-    m_pShader->use(&m_position, &m_color);
-
     // draw
+    if (withTexture) {
+        GLuint textureID = texture->getId();
+        ((ShaderTexture *)m_pShader)->use(&m_position, textureID);
+        GLint m_attributeTexCoord = ((ShaderTexture *)m_pShader)->getAttributeTexCoord();
+        glEnableVertexAttribArray(m_attributeTexCoord);
+        glVertexAttribPointer(m_attributeTexCoord, 2, GL_FLOAT, GL_FALSE, 0, m_texCoords);
```

Comment:
these gl operations should be done in TextureShader class. It is related to TextureShader

Reply:
Why do we need to do outside of the class then ? I don't understand your answer

Reply:
Ok, now the classes Car, House and Street use the ShaderTexture class for gl texture operations. Should I move the same gl operations glEnableVertexAttribArray glVertexAttribPointer, which are used for vertices out of those classes into the ShaderBase class as well? I think, this would also make sense, although this was not done this way in the previous benchmark versions.